nomos-services/key-management-system/src/backend/preload.rs

```diff
@@ -0,0 +1,196 @@
+use std::collections::HashMap;
+
+use bytes::Bytes;
+use either::Either;
+use overwatch_rs::DynError;
+use serde::{Deserialize, Serialize};
+use zeroize::ZeroizeOnDrop;
+
+use crate::{keys::ed25519::Ed25519Key, secure_key::SecuredKey, KMSOperator};
+
+use super::KMSBackend;
+
+pub struct PreloadKMSBackend {
+    keys: HashMap<String, Key>,
+}
+
+/// This settings contain all [`Key`]s to be loaded into the [`PreloadKMSBackend`].
+/// This implements [`serde::Serialize`] for users to populate the settings from bytes.
+/// The [`Key`] also implements [`zeroize::ZeroizeOnDrop`] for security.
+#[derive(Serialize, Deserialize)]
+pub struct PreloadKMSBackendSettings {
+    keys: HashMap<String, Key>,
+}
+
+#[async_trait::async_trait]
+impl KMSBackend for PreloadKMSBackend {
+    type SupportedKeys = SupportedKeys;
+    type KeyId = String;
+    type Settings = PreloadKMSBackendSettings;
+
+    fn new(settings: Self::Settings) -> Self {
+        Self {
+            keys: settings.keys,
+        }
+    }
+
+    /// This function just checks if the key_id was preloaded successfully.
+    /// This backend doesn't support generating a new key based on the [`SupportedKeys`] provided,
+    /// because this backend preloads all keys from the settings.
+    /// In other words, the users can execute operations with the key_id
+    /// even without the need to register it, if the key was specified in the settings.
```

Comment:
But what if a service wasn't given an id for the key? How would it ask for it the first time?
As this is a contract we should use the established contract (trait) in the services and it should work the same. In the case of registering, instead of returning an error, I would return the corresponding keyid.

Reply:
This comment is related to the other comment below https://github.com/logos-co/nomos-node/pull/992#discussion_r1926855623. It depends on how we define the `SupportedKeys`.